### PR TITLE
#167641775 List All Users 

### PR DESCRIPTION
**Pull request by mutoromike**
## Description 

  - Get all users and their details fro the API.

## Fix

  - Created a class handling the functionality.
  - Integrated marshmallows for the new requirement

## How to test 

- Clone the repository, install docker, run `make test` within the repository.
- Trigger a build on CircleCI.

## Screenshots

Screenshot 2019-08-09 at 12 07 27

### CODE REVIEW

**Review by kevinbett — APPROVED** [review]
Looks good @mutoromike

File: src/tests/test_user_information.py [hunk 1]
```diff
@@ -22,12 +22,17 @@ class UserInformationTestCase(BaseTestCase):
 
     def setUp(self):
         """Set up patch information for every test."""
-        super().setUp()
+        # super().setUp()
```
Comment (frankip) [inline src/tests/test_user_information.py]: Remove cemented code out

**Review by frankip — APPROVED** [review]
LGTM

**Review by georgreen — CHANGES_REQUESTED** [review]
Hey, the logic looks good but I must reject this design as it's not restful(adding an endpoint `user\all`), we already have a user Resource and this would be best implemented there.  

[Here](https://github.com/andela/andela-societies-backend/blob/ffb5989ec07f894939515254c40423cd28e3fb20/src/api/endpoints/users/__init__.py#L13)

I'm interested to know what design choices lead to this particular implementation.

**Comment by mutoromike** [comment]
@georgreen, thanks for the review. The resource you are referring to, (`UserAPI`), is tailored to return a `single user's data` given the user id. The same resource already has a get method (which implements the functionality). The admin feature needs an endpoint to return a list of users.

We thought it would be a better implementation having a different endpoint returning `all users` since as an `ADMIN`, one would need to view a list of users for them to assign, edit and revoke users roles.

**Comment by georgreen** [comment]
> @georgreen, thanks for the review. The resource you are referring to, (`UserAPI`), is tailored to return a `single user's data` given the user id. The same resource already has a get method (which implements the functionality). The admin feature needs an endpoint to return a list of users.
> 
> We thought it would be a better implementation having a different endpoint returning `all users` since as an `ADMIN`, one would need to view a list of users for them to assign, edit and revoke users roles.

You can add a `UserAll`  resource with Url `/users/` to this blueprint, check out logged points as an example, it has multiple resources [Here](https://github.com/andela/andela-societies-backend/blob/ffb5989ec0/src/api/endpoints/logged_activities/__init__.py)

**Comment by mutoromike** [comment]
> > @georgreen, thanks for the review. The resource you are referring to, (`UserAPI`), is tailored to return a `single user's data` given the user id. The same resource already has a get method (which implements the functionality). The admin feature needs an endpoint to return a list of users.
> > We thought it would be a better implementation having a different endpoint returning `all users` since as an `ADMIN`, one would need to view a list of users for them to assign, edit and revoke users roles.
> 
> You can add a `UserAll` resource with Url `/users/` to this blueprint, check out logged points as an example, it has multiple resources [Here](https://github.com/andela/andela-societies-backend/blob/ffb5989ec0/src/api/endpoints/logged_activities/__init__.py)

That's exactly what I've done. There are two resources under this blueprint; `UserAPI` and `UsersAPI`. With `UsersAPI` returning all users when the endpoint `/users/all/` is hit.

**Pull request merged**